src/java/org/httpkit/client/Decoder.java

```diff
@@ -144,10 +142,21 @@ private void readBody(ByteBuffer buffer, State nextState) throws AbortException
         }
     }
 
-    void readEmptyLine(ByteBuffer buffer) {
-        byte b = buffer.get();
-        if (b == CR && buffer.hasRemaining()) {
-            buffer.get(); // should be LF
+    private void readEmptyLine(ByteBuffer buffer, State nextState) throws ProtocolException {
+        boolean more = true;
+        while (buffer.hasRemaining() && more) {
+            byte b = buffer.get();
+            if (b == CR) {
+                if (buffer.hasRemaining() && buffer.get() == LF) {
```

Comment:
if `buffer.get() == LF` is false, then the byte returned from `buffer.get()` is lost.  
a CR should always be followed by a LF and if NOT, should it throw a protocolexception?

Reply:
Yeah, this is ugly and there is a lot more ugliness in the decoder. I just did a quick and dirty fix matching the logic in `LineReader.readLine()`. I'll do a more correct fix in a separate PR.